client/src/details/connection/CurlConnection.cpp

```diff
@@ -424,3 +422,11 @@ void CurlConnection::ProcessRetry(int attempt, const Result& httpResult)
     LOG_INFO(m_handler, "Sleeping for %lld ms", static_cast<long long>(retryDelay.count()));
     std::this_thread::sleep_for(retryDelay);
 }
+
+void CurlConnection::SetHeaders(CurlHeaderList& headers, const std::string cv)

```

Comment:
i brought this up in an earlier PR and i think it still applies.  Call it a nit if you want, and certainly not blocking.

This is named "CurlConnection", however this is actually a mix of behaviors and (curl) connection.

* adding user agent and CV headers are specific to SFS's expectation of the HTTP payload.
* Other HTTP connection logic is generic to HTTP.
* This is built on HTTP, but the items previously mentioned aren't tied to CURL.  it just happens that this implementation is curl-based.

Hypothetical: We introduce the need to send same headers / payload over a different protocol (MQMT?)  What could be leveraged from this class, would we need to fork to a new class?

Reply:
+1 and +1 :-)

Reply:
While I agree with the suggestion for the case where we need to add a new protocol, I think this should wait for when/if that time happens. Otherwise there is not much added value to the separation at the moment.

Reply:
Yet you separated out Connection and CurlConnection.  Why didn't you just use a CurlConnection?  My suggestion isn't blocking, as I mentioned, but I always like to know what criteria is used in similar decision points.

Reply:
For that one it was to simplify mocking on unit tests